src/bsoncxx/json.cpp

```diff
@@ -274,15 +274,16 @@ std::string to_json(types::value value) {
 
     json_visitor v(ss, false, 0);
 
-    switch ((int)value.type()) {
-#define BSONCXX_ENUM(name, val)            \
-    case val:                              \
-        v.visit_value(value.get_##name()); \
-        break;
+    if(value.type()) {
+        switch ((int)*(value.type())) {
```

Comment:
also, instead of int, please use the `std::underlying_type` type trait to get the actual type (which is currently uint8_t, but may change)

i.e.

`switch (static_cast<typename std::underlying_type<bsoncxx::type>::type>(*value.type()) {`